pkg/raft/raft.go

```diff
@@ -1718,6 +1742,8 @@ func (r *raft) handleSnapshot(m pb.Message) {
 	if r.restore(s) {
 		r.logger.Infof("%x [commit: %d] restored snapshot [index: %d, term: %d]",
 			r.id, r.raftLog.committed, sindex, sterm)
+		// log is now consistent with leader
+		r.accTerm = sterm
```

Comment:
I think this needs to be `m.Term`, same as in the append case. We're saying here that the state of the log/snapshot is now consistent with the leader.

I suspect `sterm == m.Term` (because the leader always sends a snapshot including at least one of its entries), but semantically we should better use `m.Term` (in case the snapshot is sent by someone else, or is outdated).

Reply:
agreed it should be the term of the raft message instead of term on the snapshot.

> in case the snapshot is sent by someone else, or is outdated
just curious from reading the raft paper if its outdated  or its not sent from a quorum leader the restore will fail right?

Reply:
If it's not sent by the current `Term` leader (or on behalf of it), the snapshot will be [ignored](https://github.com/cockroachdb/cockroach/blob/0144c31513c1ae5cdef4d78a051c17e40813f34d/pkg/raft/raft.go#L1053-L1106).

However, we can't generally assume that the snapshot `(index, term)` has the same `Term`. Say, the leader's `Term` is 2, and log entries terms are `[1 1 1 2 2 2]`, and the leader has only applied/committed the first 3 entries. Then the snapshot will be for `(index=3, term=1)`.